fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -200,6 +206,54 @@ fun IconListItem(
     )
 }
 
+/**
+ * List item used to display a label with an optional description text and
+ * an [RadioButton] at the beginning.
+ *
+ * @param label The label in the list item.
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param description An optional description text below the label.
+ * @param maxDescriptionLines An optional maximum number of lines for the description text to span.
+ * @param onClick Called when the user clicks on the item.
+ * @param selected [Boolean] That indicates whether the [RadioButton] is currently selected.
+ */
+@Composable
+fun RadioButtonListItem(
```

Comment:
as part of this PR, or in an immediate follow-up, it might make sense to add a top-level component (in a new file) for handling a group of radio button list items and their selection logic.